File: programacao-orientada-a-objetos/t2/c++/code/source/core/Entity.cc
#include "../../include/core/Entity.hh"

Entity::Entity() {
	this->id = new String();
}

Entity::Entity(char *id) {
	this->id = new String(id);
}

Entity::Entity(String *id) {
	this->id = new String(id);
}

Entity::~Entity() {
	delete this->id;
}

void Entity::setId(String *id) {
	this->id->set(id);
}

String *Entity::getId() {
	return this->id->clone();
}

void Entity::sendMessage(String *msg) {
	
}

void Entity::sendMessage(Image *img) {
	
}

void Entity::sendMessage(unsigned char *audio) {
	
}

File: programacao-orientada-a-objetos/t3-branch/src/msgr/core/Entity.java
package msgr.core;

public class Entity {
	protected int id;
	
	public Entity() { this.id = 0; }
	public Entity(int id) { this.setId(id); }
	public void setId(int id) { this.id = id; }
	public int getId() { return this.id; }
}
